Stat Calculator

slabserif opened the pull request:
Started laying ground work for how a pokemon's stats are initially
created, and how they are modified upon levelling up.

slabserif committed:
Stat Calculator

Started laying ground work for how a pokemon's stats are initially
created, and how they are modified upon levelling up.

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+        //Dv reserach is derived from: http://bulbapedia.bulbagarden.net/wiki/Individual_values
+        /*static int DvRandomizer(int min, int max)
+        {
+            Random random = new Random();
```
we will make this a global instead of recreating it every time

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+        /*static int DvRandomizer(int min, int max)
+        {
+            Random random = new Random();
+            int dv = random.Next(0, 15);
```
15?

slabserif replied:
Should be changed to 16 since it's an exclusive maximum

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+        //--Is the stat calculation meant to add on to an existing stat, or replace it?
+        //--How do Exp points fit into this and what determines how much exp is needed per level?
+
+        //Dv randomizer is used in generating the pokemon's initial stats, as well as making levelling up produce diverse stats between two of the same pokemon at the same level.
```
dv?

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+            return dv;
+        }
+
+        public int CalcNewStat(int pokemon.basestat, int dv)
```
just call it base stat

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+        public int CalcNewStat(int pokemon.basestat, int dv)
+        {
+            decimal newStat = ((((basestat + dv) * 2) + ((Math.Sqrt(Ev) / 4) * level) / 100) + 5;
+            //TODO: round down decimal newHp, and convert to int
```
Math.Round will do that

Gekctek commented on src/Core/PokemanzUtil.cs:
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
+
+        public int CalcNewHp(int pokemon.hp, int dv)
+        {
+            decimal newHp = ((((basestat + dv) * 2) + ((Math.Sqrt(Ev) / 4) * level) / 100) + level + 10;
```
looks like a copy and paste job, combine them

Gekctek commented on src/Core/Pokemon.cs:
```diff
@@ -21,6 +21,7 @@ public class Pokemon
         public int SpAttack { get; set; }
         public int SpDefense { get; set; }
         public int Speed { get; set; }
+        public int Ev { get; set; }
```
ev?

Gekctek commented on src/Core/compiler/resources/pokedex.txt:
```diff
@@ -1,11 +1,11 @@
 Id	Name	Type 1	Type 2	Level	Hp	Attack	Defense	SpAttack	SpDefense	Speed
-1	Bulbasuar	Grass		5	21	12	11	11	13	11
+1	Bulbasuar	Grass			45	49	49	65	65	45
```
why are the stats so much bigger for this one

slabserif replied:
supposedly actual base stats. Doesn't seem right, need to research more.

Gekctek commented:
I will pull this once it is fixed and when something is acutally done. as long as it is in a branch and pushed/sync'd with the server it is good. only pull request when it is done

slabserif committed:
Combined Stat/Hp Methods

Combined methods to calculate new hp and stats on level up.
Conditionally finished formula based on if hp or other stat. Rounds down
result. Turns result into an int and returns. Cannot test, do not have
all required variables available.

slabserif committed:
Added exp calculators

-Added two exp calculators to determine how much exp a pokemon needs to
gain the next level.
-Added excel column with exp type that is passed into the exp
calculators to know which formula to use
-Added excel column expgiven which is the base exp a pokemon will earn
from defeating a certain pokemon (this base exp will be plugged into a
calculation that has not been written yet.)

slabserif committed:
Battle win exp add

Created method for adding a defeated pokemons exp value to the winning
pokemons exp count. Ifthe exp count exceeds the amount needed to level
up, the level up method is run.

slabserif committed:
Stat exp calc and reset

Added method PokemonDefeatStatExp() which takes the defeated pokemon's
base stat of each stat and calculates the amount of points given to the
winning pokemon for each of their stats. This is added to a stored value
that is then used on level up to determine how much a pokemons stats get
to increase. Method ResetStatExp sets the stored value back to 0 on
level up.

slabserif committed:
Added ass-ets

Spritesheets added. Ethan cleaned up code and i pretended i understood
what was going on,

slabserif committed:
Fixed all bugs

Bugs squashed

Commit:
Changing spaces to tabs

Commit:
changing excel repository for new stat type

slabserif committed:
Added exp gain method

Add experience from defeated pokemon to used pokemon

slabserif committed:
Merge remote-tracking branch 'origin/Stat-Calculator' into Stat-Calculator

# Conflicts:
#	src/Core/PokemanzUtil.cs
#	src/Core/Pokemon.cs

slabserif committed:
Eliminated empty excel rows
